core/server/apps/private-blogging/lib/middleware.js

```diff
@@ -70,7 +70,9 @@ privateBlogging = {
         // NOTE: Redirect to /private if the session does not exist.
         privateBlogging.authenticatePrivateSession(req, res, function onSessionVerified() {
             // CASE: RSS is disabled for private blogging e.g. they create overhead
-            if (req.path.lastIndexOf('/rss/', 0) === 0 || req.path.lastIndexOf('/rss/') === req.url.length - 5) {
+            let path = req.path;
+
+            if (path === '/rss' || (path.length > 4 && path.lastIndexOf('/rss/') === path.length - 5)) {
```

Comment:
> path === '/rss'

This is not really needed. Ghost will redirect any request without trailing slash. The redirect middleware should/could kick-in before this logic happens, but this is unrelated to this PR. We should not take care of none trailing slash cases anywhere right now.

---

To be honest, i am not sure why this condition was written so complicated. Maybe there is a reason, but i don't get it. Originally added/extended [here](https://github.com/TryGhost/Ghost/commit/8a16dd5d7e7ab00bfe85ec7b7730dd6f82af21e8).

We need to ensure that all rss routes, expect of the private RSS url, result in a 404 response for private blogging. So we need to ensure blogs on subdirectories work, tag rss is covered, normal rss etc.

So I think we can simplify this condition to? 🤔 

> path.match(/\\/rss\\//)

Feel free to add any cases to `core/test/unit/apps/private-blogging/middleware_spec.js`.

Reply:
I'm quite busy now, will take a look at it soon. Did you run the tests? Tests failed so I need to wrote it that way to cover all cases.

Reply:
Yeah i did 👍

Reply:
This test will fail if I don't check for `/rss` (no trailing slash).

`it('filterPrivateRoutes should 404 for /rss requests', function () {` at line 245 in `core/test/unit/apps/private-blogging/middleware_spec.js`

Reply:
Yeah, not sure the test is 100% correct, because `/rss` will be redirected anyway to `/rss/`.

Alternative, we could use:

> path.match(/\\/rss\\/?/)